sharding/indexHashedNodesCoordinator.go

```diff
@@ -681,28 +681,41 @@ func (ihgs *indexHashedNodesCoordinator) IsInterfaceNil() bool {
 	return ihgs == nil
 }
 
-func displayNodesConfiguration(eligible map[uint32][]Validator, waiting map[uint32][]Validator, leaving []Validator, actualLeaving []Validator) {
-	for shardId, validators := range eligible {
-		for _, v := range validators {
+func (ihgs *indexHashedNodesCoordinator) displayNodesConfiguration(
+	eligible map[uint32][]Validator,
+	waiting map[uint32][]Validator,
+	leaving []Validator,
+	actualLeaving []Validator,
+	nbShards uint32,
+) {
+
+	for _, validator := range eligible[core.MetachainShardId] {
+		pk := validator.PubKey()
+		log.Debug("eligible", "pk", pk, "shardId", core.MetachainShardId)
+	}
+	for _, validator := range waiting[core.MetachainShardId] {
+		pk := validator.PubKey()
+		log.Debug("waiting", "pk", pk, "shardId", core.MetachainShardId)
+	}
+
+	for shardId := uint32(0); shardId < nbShards; shardId++ {
+		for _, v := range eligible[shardId] {
```

Comment:
you change this for to 
```go
for shard := uint32(0); shard <= nbShards; shard++ {
  shardID := shard 
  if shardID == nbShards{
    shardID = core.MetachainShardId
  }
...
}
```
then you could also remove previous two for loops for the metachain

Reply:
fixed